alicloud/resource_alicloud_cs_kubernetes.go

```diff
@@ -566,19 +636,68 @@ func resourceAlicloudCSKubernetesDelete(d *schema.ResourceData, meta interface{}
 	})
 }
 
-func buildKunernetesArgs(d *schema.ResourceData, meta interface{}) (*cs.KubernetesCreationArgs, error) {
+func isMultiAZClusterAndCheck(d *schema.ResourceData) (bool, error) {
+	masterInstanceTypes := expandStringList(d.Get("master_instance_types").([]interface{}))
+	workerInstanceTypes := expandStringList(d.Get("worker_instance_types").([]interface{}))
+	vswitchIDs := expandStringList(d.Get("vswitch_ids").([]interface{}))
+	workerNumbers := expandIntList(d.Get("worker_numbers").([]interface{}))
+
+	if len(masterInstanceTypes) != len(workerInstanceTypes) {
+		return false, errors.New("length of fields `master_instance_types`, `worker_instance_types` must match")
+	}
+	if len(masterInstanceTypes) == 1 {
+		// single AZ
+		return false, nil
+	} else if len(masterInstanceTypes) == 3 {
+		if len(vswitchIDs) != 3 || len(workerNumbers) != 3 {
+			return true, errors.New("length of fields `vswitch_ids`, `worker_numbers` must be 3 for multiAZ clusters")
+
+		} else {
+			return true, nil
+		}
+	} else {
+		return false, errors.New("length of fields `master_instance_types`, `worker_instance_types` should be either 3 (for MultiAZ) or 1 (for Single AZ)")
+	}
+}
+
+func buildKubernetesArgs(d *schema.ResourceData, meta interface{}) (*cs.KubernetesCreationArgs, error) {
 	client := meta.(*AliyunClient)
 
 	// Ensure instance_type is valid
 	zoneId, validZones, err := meta.(*AliyunClient).DescribeAvailableResources(d, meta, InstanceTypeResource)
 	if err != nil {
 		return nil, err
 	}
-	if err := meta.(*AliyunClient).InstanceTypeValidation(d.Get("master_instance_type").(string), zoneId, validZones); err != nil {
+
+	var masterInstanceType, workerInstanceType, vswitchID string
+	var workerNumber int
+
+	if list := expandStringList(d.Get("master_instance_types").([]interface{})); len(list) > 0 {
+		masterInstanceType = list[0]
```

Comment:
There is no need to condition `len(list)` because of `Required: true` and `MinItems` can ensure it.